core-java-modules/core-java-io/src/main/java/com/baeldung/file/separator/FilePathSeparator.java

```diff
@@ -0,0 +1,20 @@
+package com.baeldung.file.separator;
+
+import java.io.File;
+import java.util.StringJoiner;
+
+public class FilePathSeparator {
+
+    public static String buildPathUsingString(String[] pathNames) {
+
+        return String.join(File.pathSeparator, pathNames);
+    }
+
+    public static String buildPathUsingStringJoiner(String path4, String path5) {
```

Comment:
Why `path4` and `path5` ? They're odd variable names?

Couldn't this be implemented using

```java
buildPathUsingStringJoiner(String ... paths) {

        StringJoiner joiner = new StringJoiner(File.pathSeparator);
        for (String path : paths) {
            joiner.add(path);
        }
        return joiner.toString();
}
```